src/main/java/games/strategy/net/ClientMessenger.java

```diff
@@ -37,39 +42,42 @@
   private volatile boolean m_shutDown = false;
 
   /**
-   * Note, the name paramater passed in here may not match the name of the
+   * Note, the name parameter passed in here may not match the name of the
    * ClientMessenger after it has been constructed.
    */
   public ClientMessenger(final String host, final int port, final String name, final String mac,
       final IConnectionLogin login) throws IOException {
     this(host, port, name, mac, new DefaultObjectStreamFactory(), login);
   }
 
+
   /**
-   * Note, the name paramater passed in here may not match the name of the
+   * Note, the name parameter passed in here may not match the name of the
    * ClientMessenger after it has been constructed.
    */
+  @VisibleForTesting
   public ClientMessenger(final String host, final int port, final String name, final String mac)
       throws IOException {
-    this(host, port, name, mac, new DefaultObjectStreamFactory());
+    this(host, port, name, mac, new DefaultObjectStreamFactory(), null);
   }
 
   /**
-   * Note, the name paramater passed in here may not match the name of the
-   * ClientMessenger after it has been constructed.
-   */
-  public ClientMessenger(final String host, final int port, final String name, final String mac,
-      final IObjectStreamFactory streamFact) throws IOException {
-    this(host, port, name, mac, streamFact, null);
-  }
-
-  /**
-   * Note, the name paramater passed in here may not match the name of the
+   * Note, the name parameter passed in here may not match the name of the
    * ClientMessenger after it has been constructed.
    */
   public ClientMessenger(final String host, final int port, final String name, final String mac,
       final IObjectStreamFactory streamFact, final IConnectionLogin login)
       throws IOException {
+    Preconditions.checkNotNull(mac);
+
+    Preconditions.checkState(mac.length() == 28,
+        "incorrect mac length: " + mac.length() + ", value = " + mac);
+    final String startString = MD5Crypt.MAGIC + "MH$";
```

Comment:
This detail and the surrounding validity checks seem like good candidates to be extracted to a different class that would have responsibility for this type of thing.  Is that what you were trying to do with the `isValidMac()` method that I'm assuming you meant to add as part of this PR?

Reply:
#isValidMac logic was copy pasted between two classes, breaking up a very large branch, I missed adding that file to this branch. Beyond that, these were fixes to help debug some RMI issues where we did not fail as early as we could, and debugging was difficult

Reply:
I don't think extracting the validation logic is quite worthwhile. At the moment I think it would be extra indirection. Overall these methods could use a good bit of breaking up and extraction, I would focus efforts there rather than cleaning up opportunistic error checking.